_studio/mfx_lib/encode/mjpeg/src/mfx_mjpeg_encode.cpp

```diff
@@ -659,10 +659,11 @@ mfxStatus MFXVideoENCODEMJPEG::MJPEGENCODECompleteProc(void *pState, void *pPara
     }
 
     pTask->bs->DataLength += (mfxU32)(pDataOut.GetDataSize());
-    
+    pTask->bs->FrameType = MFX_FRAMETYPE_I;
```

Comment:
I always thought that such frame encoding characteristics as gop structure should be assigned within EncodeFrameAsync and only data is filled once SyncOperation is completed. From this perspective this assignment looks incorrect - the one which you did in EncodeFrameCheck should be enough.

Reply:
Sure, moved to `MFXVideoENCODEMJPEG::EncodeFrameCheck()`.